firmware/firmware.ino

```diff
@@ -377,17 +382,28 @@ int disable_hv(void)
  * the falling edge, but I can't be sure. */
 int set_dac(float value)
 {
-    uint16_t code = value*0x10000000000000/DAC_VREF;
+    //uint16_t code = value*0x10000000000000/DAC_VREF;
+    uint16_t voltcode = (value/DAC_VREF)*16384;
+    uint16_t code =  voltcode | 0x4000;
```

Comment:
I think you can get rid of the | 0x4000 here since we add the DAC_WRITE_THROUGH | before sending it.